src/server/main.cc

```diff
@@ -11,14 +11,40 @@
 #include "src/proto/protocached.pb.h"
 #include "src/proto/protocached.grpc.pb.h"
 
-void RunServer() {
+// TODO(joejoevictor): Move this to separate header and implementation file
+class ProtocachedServiceImpl final : public protocached::Protocached::Service {
+  grpc::Status Get(
+    grpc::ServerContext* context,
+    const ::protocached::GetRequest* request,
+    protocached::GetResponse* response
+  ) override {
+    protocached::CachedValue cachedValue;
+    cachedValue.set_value("This is a hard-coded value");
+    response -> mutable_value() -> CopyFrom(cachedValue);
+    return grpc::Status::OK;
+  } 
+  
+  grpc::Status Set(
+    grpc::ServerContext* context,
+    const ::protocached::SetRequest* request,
+    protocached::SetResponse* response
+  ) override {
+    return grpc::Status::OK;
+  }
+};
 
+void RunServer() {
+  ProtocachedServiceImpl service;
+  grpc::ServerBuilder builder;
+  builder.AddListeningPort("0.0.0.0:18081", grpc::InsecureServerCredentials());
+  builder.RegisterService(&service);
+  std::unique_ptr<grpc::Server> server(builder.BuildAndStart());
```

Comment:
why unique_ptr but not smart ptr etc? I need to read up on this.

Reply:
It's the I don't understand this part either. When I learned cpp, unique_ptr pointer isn't a thing yet. Just copying from the example so you're right we definitely have some study to do here.